core/src/test/java/org/apache/iceberg/rest/requests/TestCreateNamespaceRequest.java

```diff
@@ -0,0 +1,143 @@
+  // and can be compared with our builder after calling `validate` to normalize the parsed object.
+  public void testRoundTripSerDeAfterEnsureRequirements() throws JsonProcessingException {
+    String jsonWithNullProperties = "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":null}";
+    CreateNamespaceRequest req = CreateNamespaceRequest.builder().withNamespace(NAMESPACE).build();
+    assertDeserializesEquallyAfterValidation(jsonWithNullProperties, req);
+  }
+
+  @Test
+  public void testDeserializeInvalidRequest() {
+    String jsonIncorrectTypeForNamespace = "{\"namespace\":\"accounting%00tax\",\"properties\":null}";
+    AssertHelpers.assertThrows(
+        "A JSON request with incorrect types for fields should fail to deserialize and validate",
+        JsonProcessingException.class,
+        "Namespace must be an array",
+        () -> MAPPER.readValue(jsonIncorrectTypeForNamespace, CreateNamespaceRequest.class).validate()
+    );
+
+    String jsonIncorrectTypeForProperties = "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":[]}";
+    AssertHelpers.assertThrows(
+        "A JSON request with incorrect types for fields should fail to parse and validate",
+        JsonProcessingException.class,
+        () -> MAPPER.readValue(jsonIncorrectTypeForProperties, CreateNamespaceRequest.class).validate()
+    );
+
+    String jsonMisspelledKeys = "{\"namepsace\":[\"accounting\",\"tax\"],\"propertiezzzz\":{\"owner\":\"Hank\"}}";
+    AssertHelpers.assertThrows(
+        "A JSON request with the keys spelled incorrectly should fail to deserialize and validate",
+        JsonProcessingException.class,
+        () -> MAPPER.readValue(jsonMisspelledKeys, CreateNamespaceRequest.class).validate()
+    );
+
+    String emptyJson = "{}";
+    AssertHelpers.assertThrows(
+        "An empty JSON object should not parse into a CreateNamespaceRequest instance that passes validation",
+        ValidationException.class,
+        () -> MAPPER.readValue(emptyJson, CreateNamespaceRequest.class).validate()
+    );
+
+    AssertHelpers.assertThrows(
+        "An empty JSON request should fail to deserialize",
+        IllegalArgumentException.class,
+        () -> MAPPER.readValue((String) null, CreateNamespaceRequest.class).validate()
+    );
+  }
+
+  @Test
+  public void testHasOnlyKnownFields() {
+    // This test ensures that only the fields that are expected (from the spec) are found on the class.
+    // If new fields are added to the spec, they should be added to the ALL_FIELDS_FROM_SPEC constant.
+    CreateNamespaceRequest req = CreateNamespaceRequest.builder()
+        .withNamespace(NAMESPACE)
+        .setProperties(PROPERTIES)
+        .build();
+
+    Assertions.assertThat(req)
+        .hasOnlyFields(ALL_FIELDS_FROM_SPEC);
+  }
+
+  // For testing JSON that could be generated by our Builder
+  private void assertRoundTripSerDe(String json, CreateNamespaceRequest expected) throws JsonProcessingException {
```

Comment:
I think you can make `expected` an object and pass that to the object mapper so that these can be shared across test classes.

Reply:
I just used a templated abstract base class instead. That way the code is shareable but we don't have to cast. Let me know if you'd prefer just using `Object`.